protos/flyteidl/service/dataproxy.proto

```diff
@@ -75,11 +115,24 @@ service DataProxyService {
       description: "Creates a write-only http location that is accessible for tasks at runtime."
     };
   }
+
   // CreateDownloadLocation creates a signed url to download artifacts.
   rpc CreateDownloadLocation (CreateDownloadLocationRequest) returns (CreateDownloadLocationResponse) {
+    option deprecated = true;
```

Comment:
nit: should we also deprecate the CreateDownloadLocationRequest and CreateDownloadLocationResponse protos?